src/plugin/ipc/file/fileconnection.cpp

```diff
@@ -856,13 +860,17 @@ bool FileConnection::checkDup(int fd)
 
 int FileConnection::openFile()
 {
-  JASSERT(jalib::Filesystem::FileExists(_path)) (_path)
+  dmtcp::string phys_path_string = "";
+  const char *phys_path =  virtual_to_physical_path ?
+                           virtual_to_physical_path(_path.c_str(), phys_path_string):
+                           _path.c_str();
+  JASSERT(jalib::Filesystem::FileExists(phys_path_string)) (_path)
```

Comment:
@mossberg:  Good catch!  I assume that the right fix is to change `phys_path_string` to `phys_path` in the `JASSERT` on line 857.  If you agree, could you change it?

And I'll comment on alternative styles for `virtual_to_physical_path()` in the general comment below.